pkg/netflow/flowaggregator/aggregator.go

```diff
@@ -0,0 +1,125 @@
+package flowaggregator
+
+import (
+	"encoding/json"
+	"sync/atomic"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	"github.com/DataDog/datadog-agent/pkg/epforwarder"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	"github.com/DataDog/datadog-agent/pkg/netflow/common"
+	"github.com/DataDog/datadog-agent/pkg/netflow/config"
+)
+
+const flowAggregatorFlushInterval = 10 * time.Second
+
+// FlowAggregator is used for space and time aggregation of NetFlow flows
+type FlowAggregator struct {
+	flowIn           chan *common.Flow
+	flushInterval    time.Duration
+	flowAcc          *flowAccumulator
+	sender           aggregator.Sender
+	stopChan         chan struct{}
+	logPayload       bool
+	flushedFlowCount uint64
+}
+
+// NewFlowAggregator returns a new FlowAggregator
+func NewFlowAggregator(sender aggregator.Sender, config *config.NetflowConfig) *FlowAggregator {
+	return &FlowAggregator{
+		flowIn:        make(chan *common.Flow, config.AggregatorBufferSize),
+		flowAcc:       newFlowAccumulator(time.Duration(config.AggregatorFlushInterval) * time.Second),
+		flushInterval: flowAggregatorFlushInterval,
+		sender:        sender,
+		stopChan:      make(chan struct{}),
+		logPayload:    config.LogPayloads,
+	}
+}
+
+// Start will start the FlowAggregator worker
+func (agg *FlowAggregator) Start() {
+	log.Info("Flow Aggregator started")
+	go agg.run()
+	agg.flushLoop() // blocking call
+}
+
+// Stop will stop running FlowAggregator
+func (agg *FlowAggregator) Stop() {
+	close(agg.stopChan)
+}
+
+// GetFlowInChan returns flow input chan
+func (agg *FlowAggregator) GetFlowInChan() chan *common.Flow {
+	return agg.flowIn
+}
+
+func (agg *FlowAggregator) run() {
+	for {
+		select {
+		case <-agg.stopChan:
+			log.Info("Stopping aggregator")
+			return
+		case flow := <-agg.flowIn:
+			agg.sender.Count("datadog.newflow.aggregator.flows_received", 1, "", flow.TelemetryTags())
```

Comment:
question: This should be `netflow` right?

```suggestion
			agg.sender.Count("datadog.netflow.aggregator.flows_received", 1, "", flow.TelemetryTags())
```

Reply:
good catch